pkg/grpccatalog/grpccatalog.go

```diff
@@ -93,12 +94,19 @@ func (r *GrpcCatalog) EnsureResources(key types.NamespacedName, displayName, pub
 }
 
 // newCatalogSource returns a CatalogSource object.
-func newCatalogSource(labels map[string]string, key types.NamespacedName, displayName, publisher, namespace, address string) *olm.CatalogSource {
-	builder := new(builders.CatalogSourceBuilder).
-		WithOwnerLabel(key.Name, namespace).
-		WithMeta(key.Name, key.Namespace).
-		WithSpec(olm.SourceTypeGrpc, address, displayName, publisher)
-
+func newCatalogSource(labels map[string]string, key types.NamespacedName, displayName, publisher, namespace, address, owner string) *olm.CatalogSource {
+	var builder *builders.CatalogSourceBuilder
+	if owner == v1.OperatorSourceKind {
+		builder = new(builders.CatalogSourceBuilder).
+			WithOpsrcOwnerLabel(key.Name, namespace).
+			WithMeta(key.Name, key.Namespace).
+			WithSpec(olm.SourceTypeGrpc, address, displayName, publisher)
+	} else if owner == v2.CatalogSourceConfigKind {
+		builder = new(builders.CatalogSourceBuilder).
+			WithCscOwnerLabel(key.Name, namespace).
+			WithMeta(key.Name, key.Namespace).
+			WithSpec(olm.SourceTypeGrpc, address, displayName, publisher)
```

Comment:
You are defeating the purpose of a builder. Create the builder with the common elements and the `if else` should only set `WithOpsrcOwnerLabel` or  `WithCscOwnerLabel`